### PR TITLE
[WordGame-01] Implemented game methods and logic

### PR DESCRIPTION
**Pull request by GilCND**
Implemented gameplay
Implemented Logic

### CODE REVIEW

File: WordGame/ViewController.swift [hunk 1]
```diff
@@ -7,13 +7,129 @@
...
+            if isOriginal(word: lowerAnswer){
+                if isReal(word: lowerAnswer){
+                    if isValid(word: lowerAnswer) {
+                        usedWords.insert(lowerAnswer, at: 0)
+                        let indexPath = IndexPath (row: 0, section: 0)
+                        tableView.insertRows(at: [indexPath], with: .automatic)
+                        return
+                    }
+                } else {
+                    showErrorMessage (errorCode: 3)
```
Comment (mofarajmandi) [inline WordGame/ViewController.swift]: Very minor styling:
```suggestion
                    showErrorMessage(errorCode: 3)
```

File: WordGame/ViewController.swift [hunk 2]
```diff
@@ -7,13 +7,129 @@
...
+            self?.submit(answer)
+        }
+        alertController.addAction(submitAction)
+        present(alertController, animated: true)
+    }
+    
+    private func submit (_ answer: String) {
+        let lowerAnswer = answer.lowercased()
+        if isPossible(word: lowerAnswer){
+            if isOriginal(word: lowerAnswer){
```
Comment (mofarajmandi) [inline WordGame/ViewController.swift]: Very minor: styling - Add Space between condition and bracket `{`
```suggestion
            if isOriginal(word: lowerAnswer) {
```

File: WordGame/ViewController.swift [hunk 3]
```diff
@@ -7,13 +7,129 @@
...
+                if isReal(word: lowerAnswer){
```
Comment (mofarajmandi) [inline WordGame/ViewController.swift]: same for here and other lines

File: WordGame/ViewController.swift [hunk 4]
```diff
@@ -7,13 +7,129 @@
...
+                }
+            } else {
+                showErrorMessage (errorCode: 2)
+            }
+        } else {
+            showErrorMessage (errorCode: 1)
+        }
+    }
+    
+    private func isPossible (word: String) -> Bool {
```
Comment (mofarajmandi) [inline WordGame/ViewController.swift]:
```suggestion
    private func isPossible(word: String) -> Bool {
```

File: WordGame/ViewController.swift [hunk 5]
```diff
@@ -7,13 +7,129 @@
...
+        guard var tempWord = title?.lowercased() else { return false }
+        for letter in word {
+            if let position = tempWord.firstIndex(of: letter) {
+                tempWord.remove(at: position) } else { return false }
+            }
+        return true
+    }
+    
+    private func isOriginal (word: String) -> Bool {
```
Comment (mofarajmandi) [inline WordGame/ViewController.swift]:
```suggestion
    private func isOriginal(word: String) -> Bool {
```

**Review by mofarajmandi — APPROVED** [review]
Looks good, just some very minor styling.

File: WordGame/ViewController.swift (hunk 2 above)
Comment (GilCND) [reply WordGame/ViewController.swift]: No problem, I will fix it.

**Pull request merged**